From e97aaa90d4e4feae0cb5ce588f7a9b0ffe40a9f3 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Mon, 19 Feb 2024 18:10:12 +0100
Subject: [PATCH 0616/1695] plat-stm32mp2: conf: support STM32MP21x SoC family

Add support for the STM32MP21x SoC family.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 2a2a41e8d2951c5598f935287d44e89e22c783cb plat-stm32mp2: conf: support STM32MP21x SoC family]
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: I70729b8df508e5e10c7bd77f5f7d17afca5d9342
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/362586
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/plat-stm32mp2/conf.mk | 17 ++++++++++++++---
 1 file changed, 14 insertions(+), 3 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp2/conf.mk b/core/arch/arm/plat-stm32mp2/conf.mk
index 88182b072..8c6496f42 100644
--- a/core/arch/arm/plat-stm32mp2/conf.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.mk
@@ -7,8 +7,10 @@ flavorlist-MP25 = $(flavor_dts_file-257F_DK) \
 
 flavorlist-MP23 = $(flavor_dts_file-235F_DK)
 
+flavorlist-MP21 = #empty
+
 # List of all DTS for this PLATFORM
-ALL_DTS = $(flavorlist-MP25) $(flavorlist-MP23)
+ALL_DTS = $(flavorlist-MP25) $(flavorlist-MP23) $(flavorlist-MP21)
 
 # Check if device-tree exist in OP-TEE source code, else search it in external
 # device tree repository
@@ -37,15 +39,24 @@ endif
 ifneq ($(filter $(CFG_EMBED_DTB_SOURCE_FILE),$(flavorlist-MP23)),)
 $(call force,CFG_STM32MP23,y)
 endif
+ifneq ($(filter $(CFG_EMBED_DTB_SOURCE_FILE),$(flavorlist-MP21)),)
+$(call force,CFG_STM32MP21,y)
+endif
 
 # CFG_STM32MP2x switches are exclusive.
 # - CFG_STM32MP25 is enabled for STM32MP25x-* targets (default)
 # - CFG_STM32MP23 is enabled for STM32MP23x-* targets
-ifeq ($(CFG_STM32MP23),y)
+# - CFG_STM32MP21 is enabled for STM32MP21x-* targets
+ifeq ($(CFG_STM32MP21),y)
+$(call force,CFG_STM32MP23,n)
+$(call force,CFG_STM32MP25,n)
+else ifeq ($(CFG_STM32MP23),y)
+$(call force,CFG_STM32MP21,n)
 $(call force,CFG_STM32MP25,n)
 else
-$(call force,CFG_STM32MP25,y)
+$(call force,CFG_STM32MP21,n)
 $(call force,CFG_STM32MP23,n)
+$(call force,CFG_STM32MP25,y)
 endif
 
 include core/arch/arm/cpu/cortex-armv8-0.mk
-- 
2.47.2

